src/subscription.ts

```diff
@@ -1,260 +1,350 @@
+import { IncomingHttpHeaders } from 'http';
+import Stripe from 'stripe';
+import { DBOS } from '@dbos-inc/dbos-sdk';
 import axios from 'axios';
+import { KnexDataSource } from '@dbos-inc/knex-datasource';
+import Fastify from 'fastify';
+import rawBodyPlugin from 'fastify-raw-body';
+import { fastifyJwtJwks } from 'fastify-jwt-jwks';
+import cors from '@fastify/cors';
 
+const DBOSDomain = process.env.APP_DBOS_DOMAIN;
+const DBOSProStripePrice = process.env.STRIPE_DBOS_PRO_PRICE ?? '';
 const stripe = new Stripe(process.env.STRIPE_SECRET_KEY as string);
+export const dataSource = new KnexDataSource('knex-ds', { client: 'pg', connection: process.env.DBOS_DATABASE_URL });
+
+export const DBOSLoginDomain = DBOSDomain === 'cloud.dbos.dev' ? 'login.dbos.dev' : 'dbos-inc.us.auth0.com';
 
-const DBOSDomain = process.env.APP_DBOS_DOMAIN;
-const DBOSLoginDomain = DBOSDomain === "cloud.dbos.dev" ? "login.dbos.dev" : "dbos-inc.us.auth0.com";
-const DBOSProStripePrice = process.env.STRIPE_DBOS_PRO_PRICE ?? "";
 const DBOSPlans = {
   free: 'free',
   pro: 'pro',
 };
 
-export class Utils {
-  // Workflow to process Stripe events sent to the webhook endpoint
-  @Workflow()
-  static async stripeEventWorkflow(ctxt: WorkflowContext, subscriptionID: string, customerID: string) {
-    // Retrieve the updated subscription from Stripe
-    const status = await ctxt.invoke(Utils).getSubscriptionStatus(subscriptionID);
-
-    // Map the Stripe customer ID to DBOS Cloud user ID
-    const dbosAuthID = await ctxt.invoke(Utils).findAuth0UserID(customerID);
-
-    // Send a request to the DBOS Cloud admin API to change the user's subscription status
-    switch (status) {
-      case 'active':
-      case 'trialing':
-        await ctxt.invoke(Utils).updateCloudEntitlement(dbosAuthID, DBOSPlans.pro);
-        break;
-      case 'canceled':
-      case 'unpaid':
-      case 'paused':
-        await ctxt.invoke(Utils).updateCloudEntitlement(dbosAuthID, DBOSPlans.free);
-        break;
-      default:
-        ctxt.logger.info(`Do nothing for ${status} status.`);
-    }
-  }
+// Register all DBOS workflows and steps
+export const stripeEventWorkflow = DBOS.registerWorkflow(stripeEventWorkflowImpl, { name: 'stripeEventWorkflow' });
+export const createSubscription = DBOS.registerWorkflow(createSubscriptionImpl, { name: 'createSubscription' });
+export const createStripeCustomerPortal = DBOS.registerWorkflow(createStripeCustomerPortalImpl, {
+  name: 'createStripeCustomerPortal',
+});
 
-  // Workflow to create a Stripe checkout session
-  @Workflow()
-  static async createSubscription(ctxt: WorkflowContext, auth0UserID: string, userEmail: string, successUrl: string, cancelUrl: string): Promise<string|null> {
-    // First, look up the customer from the accounts table
-    let stripeCustomerID = await ctxt.invoke(Utils).findStripeCustomerID(auth0UserID);
+const updateCloudEntitlement = DBOS.registerStep(updateCloudEntitlementImpl, {
+  name: 'updateCloudEntitlement',
```

Comment:
Is specifying step names mandatory?

Reply:
No but I want the name to be `updateCloudEntitlement` not `updateCloudEntitlementImpl`...